core/trino-main/src/main/java/io/trino/execution/SetTimeZoneTask.java

```diff
@@ -0,0 +1,117 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.execution;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.util.concurrent.ListenableFuture;
+import io.airlift.slice.Slice;
+import io.trino.execution.warnings.WarningCollector;
+import io.trino.metadata.Metadata;
+import io.trino.security.AccessControl;
+import io.trino.spi.TrinoException;
+import io.trino.sql.planner.ExpressionInterpreter;
+import io.trino.sql.tree.Expression;
+import io.trino.sql.tree.IntervalLiteral;
+import io.trino.sql.tree.SetTimeZone;
+import io.trino.sql.tree.StringLiteral;
+import io.trino.transaction.TransactionManager;
+
+import javax.inject.Inject;
+
+import java.util.List;
+import java.util.TimeZone;
+
+import static com.google.common.util.concurrent.Futures.immediateFuture;
+import static io.trino.SystemSessionProperties.TIME_ZONE_ID;
+import static io.trino.spi.StandardErrorCode.INVALID_LITERAL;
+import static io.trino.spi.type.TimeZoneKey.getTimeZoneKey;
+import static io.trino.spi.type.TimeZoneKey.getTimeZoneKeyForOffset;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static io.trino.type.IntervalDayTimeType.INTERVAL_DAY_TIME;
+import static io.trino.util.Failures.checkCondition;
+
+public class SetTimeZoneTask
+        implements DataDefinitionTask<SetTimeZone>
+{
+    @Inject
+    public SetTimeZoneTask()
+    {
+    }
+
+    @Override
+    public String getName()
+    {
+        return "SET TIME ZONE";
+    }
+
+    @Override
+    public ListenableFuture<?> execute(
+            SetTimeZone statement,
+            TransactionManager transactionManager,
+            Metadata metadata,
+            AccessControl accessControl,
+            QueryStateMachine stateMachine,
+            List<Expression> parameters,
+            WarningCollector warningCollector)
+    {
+        String timeZoneId = statement.getTimeZone()
+                .map(timeZone -> getTimeZoneId(metadata, accessControl, stateMachine, timeZone))
+                .orElse(TimeZone.getDefault().getID());
+        stateMachine.addSetSessionProperties(TIME_ZONE_ID, timeZoneId);
+
+        return immediateFuture(null);
+    }
+
+    private String getTimeZoneId(Metadata metadata, AccessControl accessControl, QueryStateMachine stateMachine, Expression timeZone)
+    {
+        if (timeZone instanceof IntervalLiteral) {
+            return getTimeZoneId(metadata, accessControl, stateMachine, (IntervalLiteral) timeZone);
+        }
+        else if (timeZone instanceof StringLiteral) {
```

Comment:
else is redundant here

Reply:
Could you please elaborate why do you think that `else` is redundant?
I still see its usefulness.

Reply:
because the `if` conditions are exclusive and the previous branch always returns.